main.go

```diff
@@ -1,39 +1,64 @@
 package gorillaz
 
 import (
+	"errors"
 	"fmt"
 	"github.com/gorilla/mux"
 	"github.com/skysoft-atm/gorillaz/stream"
 	"github.com/spf13/viper"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/balancer/roundrobin"
 	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/resolver"
 	"net"
 	"net/http"
+	"strconv"
+	"strings"
 	"time"
 )
 
 var initialized = false
 
 type Gaz struct {
-	Router *mux.Router
+	Router             *mux.Router
+	ServiceDiscovery   ServiceDiscovery
+	StreamProviderPort int
 	// use int32 because sync.atomic package doesn't support boolean out of the box
 	isReady        *int32
 	isLive         *int32
 	grpcServer     *grpc.Server
 	streamRegistry *streamRegistry
+	httpListener   *net.Listener
+	httpPort       int
+	context        map[string]interface{}
+}
+
+type Option func(*Gaz) error
+
+func WithContext(context map[string]interface{}) Option {
+	return func(g *Gaz) error {
```

Comment:
Do we really need this one?
I don't know how to use this Context anyway, passing magical map[string]interface{} is not usable for me

Reply:
I though you might be using it, but if no one uses it, then let's remove it.

Reply:
I see that it is used in tests to specify a different config file

Reply:
this was replaced by the WithConfigPath option